dlt/destinations/impl/lancedb/lancedb_client.py

```diff
@@ -0,0 +1,567 @@
+        self.create_record(properties, VersionSchema, version_table_name)
+
+    def create_record(
+        self, record: DictStrAny, lancedb_model: TLanceModel, table_name: str
+    ) -> None:
+        """Inserts a record into a LanceDB table without a vector.
+
+        Args:
+            record (DictStrAny): The data to be inserted as payload.
+            table_name (str): The name of the table to insert the record into.
+            lancedb_model (LanceModel): Pydantic model to parse records.
+        """
+        try:
+            tbl = self.db_client.open_table(self.make_qualified_table_name(table_name))
+        except FileNotFoundError:
+            tbl = self.db_client.create_table(self.make_qualified_table_name(table_name))
+        except Exception:
+            raise
+
+        tbl.add(lancedb_model(**record))
+
+    def get_stored_state(self, pipeline_name: str) -> Optional[StateInfo]:
+        """Loads compressed state from destination storage by finding a load ID that was completed."""
+        while True:
+            try:
+                state_table_name = self.make_qualified_table_name(self.schema.state_table_name)
+                state_records = (
+                    self.db_client.open_table(state_table_name)
+                    .search()
+                    .where(f'pipeline_name = "{pipeline_name}"')
+                    .limit(10)
+                    .to_list()
+                )
+                if len(state_records) == 0:
+                    return None
+                for state in state_records:
+                    load_id = state["_dlt_load_id"]
+                    loads_table_name = self.make_qualified_table_name(self.schema.loads_table_name)
+                    load_records = (
+                        self.db_client.open_table(loads_table_name)
+                        .search()
+                        .where(f'load_id = "{load_id}"')
+                        .to_list()
+                    )
+                    if len(load_records) > 0:
+                        state["dlt_load_id"] = state.pop("_dlt_load_id")
+                        return StateInfo(**state)
+            except Exception:
+                raise
+
+    def get_stored_schema_by_hash(self, schema_hash: str) -> StorageSchemaInfo:
+        try:
+            table_name = self.make_qualified_table_name(self.schema.version_table_name)
+            response = (
+                self.db_client[table_name]
+                .search()
+                .where(f'version_hash = "{schema_hash}"')
+                .limit(1)
+            )
+            record = response.to_list()[0]
+            return StorageSchemaInfo(**record)
+        except Exception:
+            raise
+
+    def get_stored_schema(self) -> Optional[StorageSchemaInfo]:
+        """Retrieves newest schema from destination storage."""
+        try:
+            version_table_name = self.make_qualified_table_name(self.schema.version_table_name)
+            response = (
```

Comment:
same as above, you need to sort to make the newest one be the first, otherwise this will not work properly